From 416a3ec2f83c903c13e61a8e4594c2929b3f7a1f Mon Sep 17 00:00:00 2001
From: Ruslan Bay <67730802+ruslanbay@users.noreply.github.com>
Date: Tue, 17 Feb 2026 06:27:24 +0000
Subject: [PATCH 50/56] media_entity_pads_init() requires SINK/SOURCE flags on
 each pad.

Set default pad directions in ipu_isys_subdev_init() before media_entity_pads_init()
---
 drivers/media/pci/intel/ipu-isys-subdev.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-isys-subdev.c b/drivers/media/pci/intel/ipu-isys-subdev.c
index 64562a479dc6..85b31027717b 100644
--- a/drivers/media/pci/intel/ipu-isys-subdev.c
+++ b/drivers/media/pci/intel/ipu-isys-subdev.c
@@ -946,6 +946,11 @@ int ipu_isys_subdev_init(struct ipu_isys_subdev *asd,
 			return -ENOMEM;
 	}
 
+	for (i = 0; i < num_sink; i++)
+		asd->pad[i].flags = MEDIA_PAD_FL_SINK;
+	for (i = num_sink; i < num_pads; i++)
+		asd->pad[i].flags = MEDIA_PAD_FL_SOURCE;
+
 	rval = media_entity_pads_init(&asd->sd.entity, num_pads, asd->pad);
 	if (rval)
 		goto out_mutex_destroy;
-- 
2.51.0

